Week 2

blckbrrynvdr opened the pull request:
**Данные для входа захардкожены
valid@email.com
correctpassword**


В ходе выполнения ТЗ2 было сделано

1. Добавлен глобальный контекст авторизации. Там, где ранее было сделано через state, теперь сделано через context
2. Компонент components/auth/Form.jsx был переделан в компонент функцию и контекст там использовался через хук useContext
3. Подключен material-ui и использована тема оформления loft. К сожалению не все стили схожи с макетом, поэтому местами пришлось перебивать через усиленные селекторы. Думаю это неверно, но нагуглить верного решения пока не сумел.
4. Добавлены типы к props там где они используются
5. Частично написаны тесты на рендер для компонентов. Пока не понял, как написать тест для проверки рендера карты, выдаёт ошибку. Так же не понимаю как написать тесты поведения на компоненты, которые используют контекст авторизации. Нужна будет помощь)
6. Добавил карту mapbox на страницу с картой, использовал ref
7. Вроде всё что сделал

По тестам, до поведения конкретных компонентов не добрался, долго боролся с components/input/Common.jsx. Уже не вижу где ошибка, пытаюсь проверить в отрендеренном компоненте атрибуты, но в них приходит null. Ошибка наверняка какая-то очень глупая, но я её не вижу.

blckbrrynvdr committed:
feat(added context and map based on ref): added auth context and map based on ref

blckbrrynvdr committed:
feat(form): added material-ui and loft theme

added material-ui and loft theme, tests form and withAuth context

blckbrrynvdr committed:
feat(auth form): rework auth context in components/auth/Form

blckbrrynvdr committed:
feat(proptypes): add propTypes to components

blckbrrynvdr committed:
test: added render tests to components

blckbrrynvdr committed:
feat(styles): added some styles at welcome page

Allarepossible commented on src/App.jsx:
```diff
@@ -1,9 +1,16 @@
+
   state = {
     page: "home",
     auth: false,
```
кажется свойство auth более не используется

blckbrrynvdr replied:
Убрал state и методы, которые более не используются

Allarepossible commented on src/components/auth/Form.jsx:
```diff
@@ -1,37 +1,57 @@
-  );
-};
+
+  const authHook = useContext(AuthContext);
```
можно не придумывать названий, а сразу нужный метод оттуда забрать:
const {logIn} = useContext(AuthContext);

blckbrrynvdr replied:
Исправил

Allarepossible commented on src/components/logo/LoftTaxi.jsx:
```diff
@@ -0,0 +1,13 @@
+
+const Logo = () => {
+  return (
+    <>
```
кажется лишняя обертка

blckbrrynvdr replied:
Убрал

Allarepossible commented on src/components/context/withAuth.jsx:
```diff
@@ -0,0 +1,39 @@
+import React from "react";
```
вообще контекст не является компонентом. лучше вынести его в отдельную папку context вне components
и файл в целом можно назвать проще, например auth

а то получается он называется как один HOC а по факту тут же еще и провайдер экспортируется

blckbrrynvdr replied:
Переделал

Allarepossible commented:
в тестах один из тестов не проходит, так как в кнопке нет атрибута:  data-testid="submitButton"

blckbrrynvdr committed:
refactor(refactoring): refactoring after codereview

I removed the extra code after introducing the context, corrected the comments after the code review

blckbrrynvdr committed:
feat(redux): trying to redux

blckbrrynvdr committed:
refactor(app.jsx reducer-example.js): remove try redux changes

blckbrrynvdr committed:
refactor(codestyle): add codestyle configs

added config files eslint,prettier

blckbrrynvdr committed:
refactor(linters): added eslint and prettier libraries

blckbrrynvdr committed:
feat(localstorage.js): add methods to work with localstorage

blckbrrynvdr committed:
feat(redux not context): replaced context with redux

blckbrrynvdr committed:
feat(auth redux): create authentication redux modules

blckbrrynvdr committed:
feat(routes): add routing with react-router-dom

blckbrrynvdr committed:
feat(auth and register): done authenticate and registration with server

blckbrrynvdr committed:
feat(profile view): add styles to profile page

blckbrrynvdr committed:
feat(card with redux): card with redux done!

blckbrrynvdr committed:
feat(auth info in store): now on reload page authorization get from store. AMAZING!!!!

blckbrrynvdr committed:
refactor(tests): fix form tests and add reducers test

blckbrrynvdr committed:
Merge pull request #3 from blckbrrynvdr/week_3

Week 3

blckbrrynvdr committed:
Revert "Week 3"

blckbrrynvdr committed:
Merge pull request #4 from blckbrrynvdr/revert-3-week_3

Revert "Week 3"

blckbrrynvdr committed:
Revert "Revert "Week 3""

blckbrrynvdr committed:
Merge pull request #5 from blckbrrynvdr/revert-4-revert-3-week_3

Revert "Revert "Week 3""
